test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -1421,3 +1513,136 @@ UCS_TEST_P(test_ucp_sockaddr_protocols, am_zcopy_64k,
+        test_ucp_sockaddr::get_test_variants_mt(variants, features,
+                                                SEND_STOP | TEST_MODIFIER_CM,
+                                                "send_stop");
+        test_ucp_sockaddr::get_test_variants_mt(variants, features,
+                                                RECV_STOP | TEST_MODIFIER_CM,
+                                                "recv_stop");
+        test_ucp_sockaddr::get_test_variants_mt(variants, features,
+                                                SEND_STOP | RECV_STOP |
+                                                TEST_MODIFIER_CM,
+                                                "bidi_stop");
+    }
+
+protected:
+    static void recv_stop(void *arg) {
+        test_ucp_sockaddr_protocols_err *test =
+            static_cast<test_ucp_sockaddr_protocols_err*>(arg);
+        test->one_sided_disconnect(test->receiver(), UCP_EP_CLOSE_MODE_FORCE);
+        while (m_err_count == 0) {
+            test->short_progress_loop();
+        }
+    }
+
+    static void send_stop(void *arg) {
+        test_ucp_sockaddr_protocols_err *test =
+            static_cast<test_ucp_sockaddr_protocols_err*>(arg);
+        test->one_sided_disconnect(test->sender(), UCP_EP_CLOSE_MODE_FORCE);
+        while (m_err_count == 0) {
+            test->short_progress_loop();
+        }
+    }
+
+    test_ucp_sockaddr_protocols_err() {
+        set_tl_timeouts(m_env);
+    }
+
+    void init() {
+        test_ucp_sockaddr_protocols::init();
+    }
+
+    void test_tag_send_recv(size_t size, bool is_exp,
+                            bool is_sync = false) {
+        /* warmup */
+        test_ucp_sockaddr_protocols::test_tag_send_recv(size, is_exp, is_sync);
+
+        /* run error-handling test */
+        test_ucp_sockaddr_protocols::test_tag_send_recv(size, is_exp, is_sync,
+                                                        (get_variant_value() &
+                                                         SEND_STOP) ?
+                                                        send_stop : NULL,
+                                                        (get_variant_value() &
+                                                         RECV_STOP) ?
+                                                        recv_stop : NULL,
+                                                        this);
+    }
+
+    static ucs_log_func_rc_t
+    warn_leak_handler(const char *file, unsigned line,
+                      const char *function, ucs_log_level_t level,
+                      const ucs_log_component_config_t *comp_conf,
+                      const char *message, va_list ap) {
+        if (level == UCS_LOG_LEVEL_WARN) {
+            std::string err_str = format_message(message, ap);
+
+            if (/* TODO: remove when tracking of UCP requests is added */
+                (err_str.find("ptr map ") != std::string::npos) ||
+                (err_str.find("was not returned to mpool ucp_requests") !=
```

Comment:
why are we OK to have leaks?

Reply:
yes, it will be fixed then by the tracking UCP requests
it is ok, since we close EP for send, we don't expect that recv request won't be completed and vice versa

Reply:
can we force-release the leaked request somehow?

Reply:
unfortunately, it could be problematic, since we need to fix C++ compilation of `ucp_request.inl` file to use `ucp_request_put()` function for force-release of UCP requests and more complex thing is to release PTR map key since it is needed in RNDV, synchronized protocols
tracking UCP requests will allow us just remove the `warn_leak_hander()` and we would expect that no warnings/errors are printed by UCP

Reply:
> can we force-release the leaked request somehow?

@yosefe done by forcibly setting COMPLETED flag and releasing a request ID if a request is in PTR MAP.